cli/vpndaemon_windows.go

```diff
@@ -0,0 +1,148 @@
+//go:build windows
+
+package cli
+
+import (
+	"io"
+	"os"
+
+	"cdr.dev/slog"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog/sloggers/sloghuman"
+	"github.com/coder/coder/v2/vpn"
+	"github.com/coder/serpent"
+)
+
+func (r *RootCmd) vpnDaemonRun() *serpent.Command {
+	var (
+		rpcReadHandleInt  int64
+		rpcWriteHandleInt int64
+		logPath           string
+	)
+
+	cmd := &serpent.Command{
+		Use:   "run",
+		Short: "Run the VPN daemon on Windows.",
+		Middleware: serpent.Chain(
+			serpent.RequireNArgs(0),
+		),
+		Options: serpent.OptionSet{
+			{
+				Flag:        "rpc-read-handle",
+				Env:         "CODER_VPN_DAEMON_RPC_READ_HANDLE",
+				Description: "The handle for the pipe to read from the RPC connection.",
+				Value:       serpent.Int64Of(&rpcReadHandleInt),
+				Required:    true,
+			},
+			{
+				Flag:        "rpc-write-handle",
+				Env:         "CODER_VPN_DAEMON_RPC_WRITE_HANDLE",
+				Description: "The handle for the pipe to write to the RPC connection.",
+				Value:       serpent.Int64Of(&rpcWriteHandleInt),
+				Required:    true,
+			},
+			{
+				Flag:        "log-path",
+				Env:         "CODER_VPN_DAEMON_LOG_PATH",
+				Description: "The path to the log file to write to.",
+				Value:       serpent.StringOf(&logPath),
+				Required:    false, // logs will also be written to stderr
+			},
+		},
+		Handler: func(inv *serpent.Invocation) error {
+			ctx := inv.Context()
+
+			if rpcReadHandleInt < 0 || rpcWriteHandleInt < 0 {
+				return xerrors.Errorf("rpc-read-handle (%v) and rpc-write-handle (%v) must be positive", rpcReadHandleInt, rpcWriteHandleInt)
+			}
+			if rpcReadHandleInt == rpcWriteHandleInt {
+				return xerrors.Errorf("rpc-read-handle (%v) and rpc-write-handle (%v) must be different", rpcReadHandleInt, rpcWriteHandleInt)
+			}
+
+			logger := inv.Logger.AppendSinks(sloghuman.Sink(inv.Stderr)).Leveled(slog.LevelDebug)
+			if logPath != "" {
+				f, err := os.Create(logPath)
+				if err != nil {
+					return xerrors.Errorf("create log file: %w", err)
+				}
+				defer f.Close()
+				logger = logger.AppendSinks(sloghuman.Sink(f))
+			}
+
+			logger.Info(ctx, "opening bidirectional RPC pipe", slog.F("rpc_read_handle", rpcReadHandleInt), slog.F("rpc_write_handle", rpcWriteHandleInt))
+			pipe, err := newBidiPipe(uintptr(rpcReadHandleInt), uintptr(rpcWriteHandleInt))
```

Comment:
Do you need to dupe the handles on windows?

Reply:
I don't think so, when the parent starts it gives all of it's handles to the child. I could duplicate it just in case though

Reply:
I looked into it a bit and duplicating won't do us much here unless we wanted to change some security/access flags on the handle but I don't think we need to worry about that at least for now